tests/test_nodestats_regression.py

```diff
@@ -0,0 +1,57 @@
+"""Regression test for nodestats tracing."""
+
+import importlib.resources as pkg_resources
+from pathlib import Path
+
+import h5py
+import yaml
+from test_io import dict_almost_equal
+
+import topostats
+from topostats.io import LoadScans, hdf5_to_dict, read_yaml
+from topostats.processing import (
+    process_scan,
+)
+
+BASE_DIR = Path.cwd()
+REGRESSION = BASE_DIR / "tests/nodestats-regression"
+
+
+def test_nodestats_tracing_results(default_config, regtest, tmp_path, load_scan_spm: LoadScans) -> None:
+    """Regression test for checking the process_scan functions correctly."""
+    load_scan_spm.get_data()
+    img_dic = load_scan_spm.img_dict
+    # Ensure there are below grains
+    default_config["grains"]["direction"] = "above"
+    _, results, img_stats = process_scan(
+        topostats_object=img_dic["minicircle"],
+        base_dir=BASE_DIR,
+        filter_config=default_config["filter"],
+        grains_config=default_config["grains"],
+        grainstats_config=default_config["grainstats"],
+        disordered_tracing_config=default_config["disordered_tracing"],
+        dnatracing_config=default_config["dnatracing"],
+        plotting_config=default_config["plotting"],
+        output_dir=tmp_path,
+    )
+    # Remove the basename column as this differs on CI
+    results.drop(["basename"], axis=1, inplace=True)
+    print(img_stats.to_string(float_format="{:.4e}".format), file=regtest)  # noqa: T201
+    print(results.to_string(float_format="{:.4e}".format), file=regtest)  # noqa: T201
+
+    # Regtest for the topostats file
+    assert Path.exists(tmp_path / "tests/resources/processed/minicircle.topostats")
+    with h5py.File(tmp_path / "tests/resources/processed/minicircle.topostats", "r") as f:
+        saved_topostats = hdf5_to_dict(f, group_path="/")
+
+    with h5py.File(REGRESSION / "minicircle.topostats.out", "r") as f:
```

Comment:
```suggestion
    with h5py.File(RESOURCES / "minicircle_nodestats.topostats", "r") as f:
```

Reply:
Resolved by moving the test into it's own PR so issues with that can be addressed there.